client/extensions/woocommerce/components/product-search/results.js

```diff
@@ -0,0 +1,55 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import {
+	areProductSearchResultsLoaded,
+	areProductSearchResultsLoading,
+} from 'woocommerce/state/sites/products/selectors';
+import { getProductSearchResults } from 'woocommerce/state/ui/products/selectors';
+import ProductItem from './item';
+
+class ProductSearchResults extends Component {
+	static propTypes = {
+		search: PropTypes.string.isRequired,
```

Comment:
missing a few props here: `isLoaded, isLoading, onSelect, products, translate` - I know we talked about props at the GM but couldn't remember if we wanted to be super explicit with them or not - I'm a fan though 😆

Reply:
I'll grab the rest of your feedback tomorrow, but for this one -- I generally only define the passed-in PropTypes (which does mean I'm still missing `onSelect`), but things like `translate` or connected props seem less useful.

Reply:
If you don't mind me adding my 2c, I'd say that strictly all cases where a component relies on a value or function _should_ be covered by either `propTypes` or `defaultProps`. If `translate` wasn't passed in we'd see runtime errors. That said, we aren't exporting this component without enhancements (`localize` & `connect`) so until we do (for testing perhaps) we could be a little less strict :)